src/main/java/org/testng/internal/TestResult.java

```diff
@@ -320,4 +336,90 @@ public boolean wasRetried() {
   public void setWasRetried(boolean wasRetried) {
     this.m_wasRetried = wasRetried;
   }
+
+  public List<ITestNGMethod> getSkipCausedBy() {
+    if (this.m_status != SKIP) {
+      return Lists.newArrayList();
+    }
+    if (skipAnalysed) {
+      return Collections.unmodifiableList(skippedDueTo);
+    }
+    skipAnalysed = true;
+    //check if there were any config failures
+    Set<ITestResult> skippedConfigs = m_context.getFailedConfigurations().getAllResults();
+    for (ITestResult skippedConfig : skippedConfigs) {
+      if (isGlobalFailure(skippedConfig) ||
+          isRelated(skippedConfig) || belongToSameGroup(skippedConfig)) {
+        skippedDueTo.add(skippedConfig.getMethod());
+        return Collections.unmodifiableList(skippedDueTo);

```

Comment:
Why only collect the first match? What if many matches?

Reply:
@juherr - Is that even possible ?

Reply:
Ok.. So there are two scenarios with configs.. 

1. `@BeforeSuite/@BeforeTest/@BeforeClass` - If one of these failed then the cause is just 1.
2. `@BeforeGroups` - If this failed then there could be more than one method. So forming a list here.